File: General Assembly WDI Boston/ruby-array-methods/spec/fibonacci_spec.rb
# frozen_string_literal: true

require 'spec_helper'

describe 'fibonacci.rb' do
  require_relative('../lib/fibonacci.rb')

  let(:fibs) { [0, 1, 1, 2, 3, 5, 8, 13, 21, 34] }

  describe '#product_of_elements' do
    it 'calculates the product of elements within fibs, excluding 0' do
      expect(product_of_elements(fibs)).to eq 2_227_680
    end
  end

  describe '#sum_of_odd_elements' do
    it 'sums the odd elements within fibs' do
      expect(sum_of_odd_elements(fibs)).to eq 44
    end
  end

  describe '#product_of_even_elements' do
    it 'calculates the product of even elements within fibs' do
      expect(product_of_even_elements(fibs)).to eq 544
    end
  end
end

File: General Assembly WDI Boston/ruby-vs-js-array-methods-study/spec/fibonacci.spec.js
'use strict'

// Allow chai syntax like `expect(foo).to.be.ok;`
// jshint -W030

const chai = require('chai')
const chaiAsPromised = require('chai-as-promised')

chai.use(chaiAsPromised)

const expect = chai.expect

const fibResults = require('../lib/fibonacci.js')

describe('subFibs', function () {
  it('is the sum of the fibs array', function () {
    expect(fibResults.sumFibs).to.equal(88)
  })
})

describe('prodFibs', function () {
  it('is the product of the fibs array excluding 0', function () {
    expect(fibResults.prodFibs).to.equal(2227680)
  })
})

describe('sumOdd', function () {
  it('is the sum of the odd numbers in the fibs array', function () {
    expect(fibResults.sumOdd).to.equal(44)
  })
})

describe('prodEven', function () {
  it('is the product of the even numbers in the fibs array excluding 0', function () {
    expect(fibResults.prodEven).to.equal(544)
  })
})
